contracts/FlowIDTableStaking.cdc

```diff
@@ -276,6 +219,12 @@ pub contract FlowIDTableStaking {
         pub let tokensRequestedToUnstake: UFix64
         pub let initialWeight: UInt64
 
+        /// Derived Fields
+        pub let totalCommittedWithDelegators: UFix64
```

Comment:
@turbolent @SupunS Will adding and removing the fields from this struct definition in the Staking contract be a problem for upgrading? The reason I thought it might be ok is that as far as I know, nobody actually stores these structs anywhere because they only provide a snapshot at a specific moment in time, so upgrading this wouldn't corrupt them, but I am worried that it will still fail the upgrade checks

Reply:
Yeah, the validation would think adding a new field is a 'potential' problem and will fail. 

The problem we are facing right now is, there is no way to detect incoming dependencies to a contract/struct (e.g: `FlowIDTableStaking` ), and no way to know whether there are any stored data or not. So assuming the worst-case scenario, we are failing such scenarios for any contract/struct.

Removing a field is fine though, because:
  - It leaves data for the removed field unused at the storage, as it is no longer accessible.
  - However, it does not cause any runtime crashes, while trying to decode.

I am working on documenting all these criteria and rationals behind the update validations. It is about 80% completed. I would be able to push it out very soon.